sensor/common/signal/component/component.go

```diff
@@ -0,0 +1,143 @@
+package component
+
+import (
+	"io"
+	"strings"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/stringutils"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+)
+
+const maxBufferSize = 10000
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type Component interface {
+	common.SensorComponent
+
+	GetReceiver() chan *sensor.ProcessSignal
+}
+
+type componentImpl struct {
+	common.SensorComponent
+
+	processPipeline Pipeline
+	indicators      chan *message.ExpiringMessage
```

Comment:
I can't see any writes to this channel here, so let's make it read only.
```suggestion
	indicators      <-chan *message.ExpiringMessage
```